openedx/core/djangoapps/credentials_service/models.py

```diff
@@ -1,3 +1,264 @@
+from django.contrib.sites.models import Site
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from model_utils.models import TimeStampedModel
+from xmodule_django.models import CourseKeyField
+
+
+MODE_CHOICES = (
+    (_('Honor'), 'honor'),
+    (_('Verified'), 'verified'),
+    (_('Professional'), 'professional'),
+)
+
+
+def template_assets_path(instance, filename):
+    """
+    Custom path for credentials templates file assets.
+
+    Arguments:
+        instance: CertificateTemplateAsset object
+        filename: file to upload
+
+    Returns:
+        Path of asset file e.g.
+        credential_certificate_template_assets/1/filename
+    """
+
+    return 'certificate_template_assets/{id}/{filename}'.format(id=instance.id, filename=filename)
+
+
+def signatory_assets_path(instance, filename):
+    """
+    Custom path for signatories file assets.
+
+    Arguments:
+        instance: Signatory object
+        filename: file to upload
+
+    Returns:
+        Path of asset file e.g.
+        signatories/1/filename
+    """
+    return 'signatories/{id}/{filename}'.format(id=instance.id, filename=filename)
+
+
+def validate_image(image):
+    """
+    Validates that a particular image is small enough.
+    """
+    if image.size > (250 * 1024):
+        raise ValidationError(_('The image file size must be less than 250KB.'))
+
+
+class AbstractCredential(TimeStampedModel):
+    """
+    Abstract Credentials configuration model.
+    """
+    site = models.ForeignKey(Site)
+    is_active = models.BooleanField(default=False)
+
+    class Meta(object):
+        abstract = True
+
+
+class Signatory(TimeStampedModel):
+    """
+    Signatory model to add certificate signatories.
+    """
+    name = models.CharField(max_length=255)
+    title = models.CharField(max_length=255)
+    image = models.ImageField(
+        help_text=_(
+            'Image must be square PNG files. The file size should be under 250KB.'
+        ),
+        upload_to=signatory_assets_path,
+        validators=[validate_image]
+    )
+
+    def __unicode__(self):
+        """Unicode representation. """
+        return '{name}, {title}'.format(
+            name=self.name,
+            title=self.title,
+        )
+
+    def save(self, *args, **kwargs):
+        """save the Signatory."""
+        if self.pk is None:
+            temp_image = self.image
+            self.image = None
+            super(Signatory, self).save(*args, **kwargs)
+            self.image = temp_image
+        else:
+            super(Signatory, self).save(*args, **kwargs)
+
+
+class CertificateTemplate(TimeStampedModel):
+    """
+    Certificate Template model to organize content for certificates.
+    """
+
+    name = models.CharField(max_length=255, db_index=True)
+    content = models.TextField(
+        help_text=_('HTML Template content data.')
+    )
+    certificate_type = models.CharField(
```

Comment:
Actually this field is not to determine whether it is `CourseCertificate` or `ProgramCertificate` this will be use to set 
mode of the certificate i.e. `Verified`, `honor` etc.